src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp

```diff
@@ -0,0 +1,186 @@
+template <typename Frame>
+bool operator!=(const ApparentHorizon<Frame>& lhs,
+                const ApparentHorizon<Frame>& rhs) noexcept;
+
+}  // namespace OptionHolders
+
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sends points on a trial apparent horizon to an `Interpolator`.
+///
+/// This differs from `KerrHorizon` in the following ways:
+/// - It supplies points on a prolonged Strahlkorper, at a higher resolution
+///   than the Strahlkorper in the DataBox, as needed for horizon finding.
+/// - It uses a `FastFlow` in the DataBox.
+/// - It has different options (including those for `FastFlow`).
+///
+/// Uses:
+/// - DataBox:
+///   - `::Tags::Domain<VolumeDim, Frame>`
+///   - `::ah::Tags::FastFlow`
+///   - `StrahlkorperTags::CartesianCoords<Frame>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// This Action also has an initialize function that adds to the DataBox:
+/// - `StrahlkorperTags::items_tags<Frame>`
+/// - `StrahlkorperTags::compute_items_tags<Frame>`
+/// - `::ah::Tags::FastFlow`
+/// - `::Tags::Verbosity`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag, typename Frame>
+struct ApparentHorizon {
+  using options_type = OptionHolders::ApparentHorizon<Frame>;
+  using const_global_cache_tags = tmpl::list<InterpolationTargetTag>;
+  using initialization_tags =
+      tmpl::append<StrahlkorperTags::items_tags<Frame>,
+                   tmpl::list<::ah::Tags::FastFlow, ::Tags::Verbosity>,
+                   StrahlkorperTags::compute_items_tags<Frame>>;
+  template <typename DbTags, typename Metavariables>
+  static auto initialize(
+      db::DataBox<DbTags>&& box,
+      const Parallel::ConstGlobalCache<Metavariables>& cache) noexcept {
+    const auto& options = Parallel::get<InterpolationTargetTag>(cache);
+
+    // Put Strahlkorper and its ComputeItems, FastFlow,
+    // and verbosity into a new DataBox.
+    return db::create_from<
+        db::RemoveTags<>,
+        db::AddSimpleTags<
+            tmpl::push_back<StrahlkorperTags::items_tags<Frame>,
+                            ::ah::Tags::FastFlow, ::Tags::Verbosity>>,
+        db::AddComputeTags<StrahlkorperTags::compute_items_tags<Frame>>>(
+        std::move(box), options.strahlkorper, options.fast_flow,
+        options.verbosity);
+  }
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent,
+            Requires<tmpl::list_contains_v<
+                DbTags, typename Tags::TemporalIds<Metavariables>>> = nullptr>
+  static void apply(
```

Comment:
I am unsure if you can test this action, because of the call to `send_points_to_interpolator`.

Reply:
Done.